core/src/main/java/org/mule/runtime/core/internal/lifecycle/phases/LifecycleObjectSorter.java

```diff
@@ -30,4 +31,11 @@
    * @return The sorted list of objects previously added through {@link #addObject(String, Object)}
    */
   List<Object> getSortedObjects();
+
+  /**
+   * Provides the order of objects as reference for initialise/dispose phases
+   * 
+   * @param lookupObjectsForLifecycle lifecycle object list which is ordered based on the type
+   */
+  void setLifeCycleObjectNameOrderMap(Map<String, Object> lookupObjectsForLifecycle);
```

Comment:
Add default empty method, so you don't have to redefine it empty in other cases later!

Reply:
done!